aiidalab_widgets_base/utils/__init__.py

```diff
@@ -114,33 +114,39 @@ class _StatusWidgetMixin(traitlets.HasTraits):
     """
 
     message = traitlets.Unicode(default_value="", allow_none=True)
+    new_line = "\n"
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, clear_after=3, *args, **kwargs):
         self._clear_timer = None
+        self._clear_after = clear_after
         self._message_stack = []
         super().__init__(*args, **kwargs)
 
     def _clear_value(self):
         """Set widget .value to be an empty string."""
         if self._message_stack:
             self._message_stack.pop(0)
-            self.value = "<br>".join(self._message_stack)
+            self.value = self.new_line.join(self._message_stack)
         else:
             self.value = ""
 
-    def show_temporary_message(self, value, clear_after=3):
+    def show_temporary_message(self, value):
         """Show a temporary message and clear it after the given interval."""
-        self._message_stack.append(value)
-        self.value = "<br>".join(self._message_stack)
+        if value:
+            self._message_stack.append(value)
+            self.value = self.new_line.join(self._message_stack)
 
-        # Start new timer that will clear the value after the specified interval.
-        self._clear_timer = threading.Timer(clear_after, self._clear_value)
-        self._clear_timer.start()
+            # Start new timer that will clear the value after the specified interval.
+            self._clear_timer = threading.Timer(self._clear_after, self._clear_value)
+            self._clear_timer.start()
+            self.message = ""
```

Comment:
I believe @csadorf won't like this, but that is the only way I could find to let the widget accept the same warning message.

Reply:
Indeed, I suspect that this could lead to some weird behavior since this function is triggered as part of the trait callback and thus would trigger the callback directly again. Maybe we can clear it together with the `.value` trait instead?

Reply:
I thought a bit about this.

> Maybe we can clear it together with the `.value` trait instead?

I am not sure what benefits it gives. Can you name any? It looks to me that the solution you propose makes a delayed trait reset. The mechanism that prevents `self.show_temporary_message()` from doing anything remains the same.

At the same time, we already have some code examples that modify the trait in the observer function and so far we didn't run into issues with such an approach.

My preferred solution would be to change the trait value but not notify the observing functions. I looked for a while and didn't find ways to temporary block the notifications. Maybe there are good reasons for it to not exist.

Reply:
The purpose of the `message` trait is to be able to trigger showing a temporary message via a trait _change_. That's a bit of a weird mechanism anyways, and maybe the `.message` trait should not exist in the first place. My main issue is that `.show_temporary_message()` both _observes and changes_ the `.message` trait. That's typically a red flag to me and could lead infinite loops. However, like you said, we are also clearing it in the `._clear_value()` function, just on a delay and we are protecting this function with the truthy check on line 136, so should be fine.

The one suggestion I have is to maybe use `None` as the explicit _null_ value since that's allowed:
```suggestion
            self.message = None
```